modules/samples/producers/jms/src/main/java/org/wso2/carbon/sample/jmsclient/JMSClient.java

```diff
@@ -49,66 +51,130 @@ public static void main(String[] args) {
+
+                }else if(broker.equalsIgnoreCase("qpid")){
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("qpid.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (QueueConnectionFactory) context.lookup("qpidConnectionFactory");
+
+                }else{
+                    log.error("Please enter a valid JMS message broker. (ex: activemq, mb, qpid");
+                }
+
+                if(connFactory != null) {
+                    queueConnection = connFactory.createQueueConnection();
+                    queueConnection.start();
+                    session = queueConnection.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
+
+                    if (session != null) {
+                        Queue queue = session.createQueue(queueName);
+                        MessageProducer producer = session.createProducer(queue);
+
+                        List<String> messagesList = JMSClientUtil.readFile(filePath);
+                        try {
+                            if (format.equalsIgnoreCase("csv")) {
+                                log.info("Sending Map messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishMapMessage(producer, session, messagesList);
+
+                            } else {
+                                log.info("Sending  " + format + " messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishTextMessage(producer, session, messagesList);
+                            }
+                        } catch (JMSException e) {
+                            log.error("Can not subscribe." + e.getMessage(), e);
+                        } catch (IOException e) {
+                            log.error("Error when reading the data file." + e.getMessage(), e);
+                        } finally {
+                            producer.close();
+                            session.close();
+                            queueConnection.stop();
+                        }
                     }
-                } catch (JMSException e) {
-                    log.error("Can not subscribe." + e.getMessage(), e);
-                } catch (IOException e){
-                    log.error("Error when reading the data file." + e.getMessage(), e);
-                }finally{
-                    producer.close();
-                    session.close();
-                    topicConnection.stop();
+                }else{
+                    log.error("Error when creating connection factory. Please check necessary jar files");
                 }
+            }else{
+                log.error("Enter queue name or topic name to be published!");
             }
         } catch (Exception e) {
-            log.error("Error when publishing message" + e.getMessage(), e);
+            log.error("Error when publishing");
```

Comment:
Thanks @prabathabey, fixed it in the pul request - https://github.com/wso2/product-cep/pull/166